core/src/msg.rs

```diff
@@ -0,0 +1,352 @@
+//! Message queue management.
```

Comment:
I've forgotten why `core/src/shm.rs` is in `core` instead of `api`, but it clearly shouldn't be there. I apologize for this being a legacy issue. However, if you're going to add a `core/src/msg.rs`, I suggest putting it directly in `api`.